### PR TITLE
Feature/auth refresh get endpoint

### PR DESCRIPTION
**Pull request by DanBaDo**
Requires valid refresh token for access.
Provides a new token based in refresh token provided in authentication.
The new token includes original refresh token. No updates in refresh token expiration.

### CODE REVIEW

File: src/http/controllers/controller.auth.ts [hunk 1]
```diff
@@ -44,13 +46,54 @@ export module AuthController {
...
+  /**
+   * ! Create new authentication tokens for valid refres tokens
+   * * DanBaDo - 2022/01/14
+   * @param req {Request}
+   * @param res {Response}
+   */
+  export const _refresh = async (req, res) => {
+    const refreshTokenString: string =  req.headers.authorization.split(' ')[1];
+    try {
+      const  refresTokeData = jwt.decode(refreshTokenString,{complete: true});
```
Comment (gaal1le0) [inline src/http/controllers/controller.auth.ts]: You are using the `jwt` library to decode a token, it's not needed, we have a middleware located in `JWTMiddelware` that you need to use to decode a token

**Review by gaal1le0 — CHANGES_REQUESTED** [review]

**Review by gaal1le0 — COMMENTED** [review]
Changes approved

**Comment by DanBaDo** [comment]
@Alcazar87 @whitehatdevv
Please, feedback.

File: src/http/controllers/controller.auth.ts (hunk 1 above)
Comment (DanBaDo) [reply src/http/controllers/controller.auth.ts]: Function removed in the comming version

**Comment by DanBaDo** [comment]
In this point, main refresh endpoint functionalities are committed.

**Pull request closed without merging**